app/src/main/java/edu/temple/templetag/fragments/MapFragment.java

```diff
@@ -37,41 +37,56 @@
 import edu.temple.templetag.HomeActivity;
 import edu.temple.templetag.R;
 import edu.temple.templetag.Tag;
+import edu.temple.templetag.tools.DistanceCalculator;
 
 public class MapFragment extends Fragment implements OnMapReadyCallback {
     private static final String ARG_PARAM1 = "tags";
+    private static final String ARG_PARAM3 = "tag";
     private static final String ARG_PARAM2 = "currentLocation";
-    private static final double MAX_RADIUS = 10;
     private Location currentLocation;
+    private DistanceCalculator distanceCalculator = new DistanceCalculator();
 
     // TODO: Rename and change types of parameters
     private ArrayList<Tag> tags;
+    private Tag tag;
 
     //Initiate Map Stuff
     MapView mapView;
-    GoogleMap googleMap;
+    public GoogleMap googleMap;
 
     public MapFragment() {
         // Required empty public constructor
     }
 
 
-    public static MapFragment newInstance() {
+    public static MapFragment newInstance(ArrayList<Tag> tags, Location currentLocation) {
```

Comment:
What are the two newInstance methods for in MapFragment? How are they used?

Reply:
The MapFragment in TagDetailActivity only needs one tag whereas MapFragment in HomeActivity needs an ArrayList of Tag.